src/D2L.CodeStyle.Analyzers/ApiUsage/ConstantAttributeAnalyzer.cs

```diff
@@ -52,6 +62,50 @@ CompilationStartAnalysisContext context
 			);
 		}
 
+		private static void AnalyzeReturn(
+			SyntaxNodeAnalysisContext context,
+			ReturnStatementSyntax returnSyntax,
+			ISymbol constantAttribute
+		) {
+			// Return statement has no expression, so do nothing
+			if( returnSyntax.Expression == null ) {
+				return;
+			}
+
+			var methodSyntax = returnSyntax.Parent?.Parent;

```

Comment:
This doesn't work... `return`s can be nested pretty deeply.

```cs
if ( ... ) {
  if ( ... ) {
    if ( ... ) {
      return ...;
```

You would need something like `.Ancestors().OfType<MethodDeclarationSyntax>()` instead.